src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -497,9 +499,27 @@ public override string MetadataName
         }
 
         /// <summary>
-        /// If this type is a file-local type, returns the syntax tree where this type is visible. Otherwise, returns null.
+        /// If this type is a file-local type, returns an identifier for the file this type was declared in. Otherwise, returns null.
         /// </summary>
-        internal abstract SyntaxTree? AssociatedSyntaxTree { get; }
+        internal abstract FileIdentifier? AssociatedFileIdentifier { get; }
+
+        internal struct FileIdentifier
+        {
+            public ImmutableArray<byte> FilePathChecksum { get; init; }
+            public string DisplayFilePath { get; init; }
+        }
+
+        public static FileIdentifier CreateFileIdentifier(SyntaxTree tree)
```

Comment:
>CreateFileIdentifier [](http://example.com/codeflow?start=37&length=20)

Should this factory method be moved into the ```FileIdentifier``` structure, and, perhaps, renamed to just "Create"?